flowmachine/tests/test_labelled_flows.py

```diff
@@ -0,0 +1,237 @@
+    loc_1 = locate_subscribers(
+        "2016-01-01",
+        "2016-01-02",
+        spatial_unit=make_spatial_unit("admin", level=3),
+        method="most-common",
+    )
+
+    loc_2 = locate_subscribers(
+        "2016-01-02",
+        "2016-01-03",
+        spatial_unit=make_spatial_unit("admin", level=2),
+        method="most-common",
+    )
+
+    labels_1 = SubscriberHandsetCharacteristic(
+        "2016-01-01", "2016-01-03", characteristic="hnd_type"
+    )
+
+    with pytest.raises(ValueError, match="same spatial unit"):
+        _ = LabelledFlows(loc1=loc_1, loc2=loc_2, labels=labels_1)
+
+
+def test_column_validation():
+    loc_1 = locate_subscribers(
+        "2016-01-01",
+        "2016-01-02",
+        spatial_unit=make_spatial_unit("admin", level=3),
+        method="most-common",
+    )
+
+    loc_2 = locate_subscribers(
+        "2016-01-02",
+        "2016-01-03",
+        spatial_unit=make_spatial_unit("admin", level=3),
+        method="most-common",
+    )
+
+    labels_1 = SubscriberHandsetCharacteristic(
+        "2016-01-01", "2016-01-03", characteristic="hnd_type"
+    )
+    with pytest.raises(ValueError, match="not present in"):
+        _ = LabelledFlows(
+            loc1=loc_1, loc2=loc_2, labels=labels_1, label_columns=["notavalue"]
+        )
+
+
+def test_subscriber_validation():
+    loc_1 = locate_subscribers(
+        "2016-01-01",
+        "2016-01-02",
+        spatial_unit=make_spatial_unit("admin", level=3),
+        method="most-common",
+    )
+
+    loc_2 = locate_subscribers(
+        "2016-01-02",
+        "2016-01-03",
+        spatial_unit=make_spatial_unit("admin", level=3),
+        method="most-common",
+    )
+
+    labels_1 = SubscriberHandsetCharacteristic(
+        "2016-01-01", "2016-01-03", characteristic="hnd_type"
+    )
+    with pytest.raises(ValueError, match="subscriber"):
+        _ = LabelledFlows(
+            loc1=loc_1, loc2=loc_2, labels=labels_1, label_columns=["subscriber"]
+        )
+
+
+def test_label_redaction(get_dataframe):
+    loc_1 = locate_subscribers(
+        "2016-01-01",
+        "2016-01-02",
+        spatial_unit=make_spatial_unit("admin", level=3),
+        method="most-common",
+    )
+
+    loc_2 = locate_subscribers(
+        "2016-01-02",
+        "2016-01-03",
+        spatial_unit=make_spatial_unit("admin", level=3),
+        method="most-common",
+    )
+
+    labels = SubscriberHandsetCharacteristic(
+        "2016-01-01", "2016-01-03", characteristic="hnd_type"
+    ).join(
+        SubscriberHandsetCharacteristic(
+            "2016-01-01", "2016-01-03", characteristic="brand"
+        ),
+        "subscriber",
+        left_append="_hnd_type",
+        right_append="_brand",
+    )
+    lf = LabelledFlows(
+        loc1=loc_1,
+        loc2=loc_2,
+        labels=labels,
+        label_columns=["value_hnd_type"],
+    )
+    assert "value_brand" not in lf.out_label_columns
+    assert "value_brand" not in get_dataframe(lf).columns
```

Comment:
If the column  _was_ there, it would be called "value_brand_label", so this test would still pass.

Perhaps better to check that only the expected columns _are_ present, rather than checking that the extra column is not.